src/app/pages/biotools-repo-page/biotools-repo-page.component.ts

```diff
@@ -386,4 +388,57 @@ export class BiotoolsRepoPageComponent {
     const downloadinfo = note.split(';');
     return downloadinfo[3].replace('config:','');
   }
+
+  private updateIcons(biotoolsApplication: BiotoolsApplication[]): void {
+
+    for(var application of biotoolsApplication) {
+      application.icons = { cloud: false, book: false, binder:false, globe: false, window: false };
+
+      console.log("Checking if application is of type Notebook");
+      if (application.toolType.length > 0 && application.toolType.indexOf('Notebook') >= 0) {
+        application.icons.book = true;
+      } else {
+      for (const d of application.download) {
+        if (d.note != null && (d.note.includes('Jupyter'))) {
+              application.icons.book = true;
+        }
+      }
+      }
```

Comment:
We are having 3 iterations over the same `downloads`. It would make sense to have it only once and check all the conditions. What do you think?

Reply:
moved to one iteration